pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -173,3 +193,54 @@ func (i *HookExecutorPodClientImpl) CreatePod(namespace string, pod *kapi.Pod) (
 func (i *HookExecutorPodClientImpl) WatchPod(namespace, name string) (watch.Interface, error) {
 	return i.WatchPodFunc(namespace, name)
 }
+
+// PodWatch provides watch semantics for a pod backed by a poller, since
+// events aren't generated for pod status updates.
+type PodWatch struct {
+	result chan watch.Event
+	stop   chan bool
+}
+
+func (w *PodWatch) Stop() {
+	w.stop <- true
+}
+
+func (w *PodWatch) ResultChan() <-chan watch.Event {
+	return w.result
+}
+
+// NewPodWatch makes a new PodWatch.
+func NewPodWatch(client kclient.Interface, namespace, name string, period time.Duration) *PodWatch {
```

Comment:
while we're refactoring this, can we delete it? :) Watching pods for status changes works now